### PR TITLE
Add details attribute to alerts table and update AlertMailer #1387

### PR DESCRIPTION
**Pull request by mattwarren10**

### CODE REVIEW

File: db/migrate/20171003172534_add_details_to_alerts.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddDetailsToAlerts < ActiveRecord::Migration[5.1]
+  def change
+  	add_column :alerts, :details, :text
```
Comment (ragesoss) [inline db/migrate/20171003172534_add_details_to_alerts.rb]: yikes, tabs!

**Comment by mattwarren10** [comment]
This is a simple way I thought of to serialize a text attribute to be a Hash. Let me know what you think.

**Comment by ragesoss** [comment]
Yes, that's what I had in mind.

**Comment by mattwarren10** [comment]
Is this something you were looking for? If it is, I'll take on another issue.

**Comment by ragesoss** [comment]
This is good so far. It still needs to:

1. Include the details in the rendered alert email (by changing the relevant view template)
2. pass in the mediawiki response data as details to new BlockedEditAlert records

The second item starts from `lib/wiki_response.rb` and goes to `BlockedEditsWorker` and `BlockedEditsReporter`.

**Comment by ragesoss** [comment]
This will be the first time we would use Redis to store something more complex than an integer record ID — passing in the details hash as a sidekiq argument in `BlockedEditsWorker` — but I think it's okay and jives with https://github.com/mperham/sidekiq/wiki/Best-Practices

That details hash is not application state, and it starts its life as plain JSON anyway, so it should survive being serialized and deserialized.

**Comment by mattwarren10** [comment]
So the check isn't passing because of the amount of arguments. Since `WikiResponse` is not a model, hence it doesn't have any factories, would it be best to just include empty hashes like this? Or is there a more optimal approach?

```
describe BlockedEditsReporter do
  describe '.create_alert' do
    let(:student) { create(:user, username: 'student') }
    let(:student2) { create(:user, username: 'student2') }
    it 'creates an Alert record for a blocked edit' do
      BlockedEditsReporter.create_alerts_for_blocked_edits(student, {})
      expect(Alert.count).to eq(1)
    end

    it 'does not create multiple alerts' do
      BlockedEditsReporter.create_alerts_for_blocked_edits(student, {})
      BlockedEditsReporter.create_alerts_for_blocked_edits(student2, {})
      expect(Alert.count).to eq(1)
    end
  end
end
```

**Comment by ragesoss** [comment]
I would pass it something more like the expected data. Here's roughly what an actual response that would get passed to WikiResponse looks like:

```ruby
{:servedby=>"mw1229",
 :error=>
  {:info=>"You have been blocked from editing.",
   :blockinfo=>
    {:blockedtimestamp=>"2017-09-16T14:12:01Z",
     :blockid=>7824827,
     :blockedby=>"Orangemike",
     :blockreason=>
      "{{uw-softerblock}} <!-- Promotional username, soft block -->",
     :blockedbyid=>42464,
     :blockexpiry=>"infinite"},
   :*=>
    "See https://en.wikipedia.org/w/api.php for API usage. Subscribe to the mediawiki-api-announce mailing list at &lt;https://lists.wikimedia.org/mailman/listinfo/mediawiki-api-announce&gt; for notice of API deprecations and breaking changes.",
   :code=>"blocked"}}

```

(I copied this from Sentry; the actual response would I think have the same shape, except with strings instead of symbols for all the keys.)

**Comment by mattwarren10** [comment]
Should be good to go.

**Pull request merged**